bindings/go/evmc/host_test.go

```diff
@@ -62,49 +62,57 @@ func (host *testHostContext) Call(kind CallKind,
 	return output, gas, Address{}, nil
 }
 
-func TestGetTxContext(t *testing.T) {
+func TestGetBlockNumberFromTxContext(t *testing.T) {
+	// Yul: mstore(0, number()) return(0, msize())
+	code := []byte("\x43\x60\x00\x52\x59\x60\x00\xf3")
+
 	vm, _ := Load(modulePath)
 	defer vm.Destroy()
 
 	host := &testHostContext{}
-	code := []byte("\x43\x60\x00\x52\x59\x60\x00\xf3")
-
 	addr := Address{}
 	h := Hash{}
 	output, gasLeft, err := vm.Execute(host, Byzantium, Call, false, 1, 100, addr, addr, nil, h, code, h)
 
-	if len(output) != 20 {
+	if len(output) != 32 {
 		t.Errorf("unexpected output size: %d", len(output))
 	}
-	if bytes.Compare(output[0:3], []byte("42\x00")) != 0 {
-		t.Errorf("execution unexpected output: %s", output)
+
+	// Should return value 42 (0x2a) as defined in GetTxContext().
+	expectedOutput := []byte("\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x2a")
+	if bytes.Compare(output, expectedOutput) != 0 {
+		t.Errorf("execution unexpected output: %x", output)
 	}
-	if gasLeft != 50 {
+	if gasLeft != 94 {
 		t.Errorf("execution gas left is incorrect: %d", gasLeft)
 	}
 	if err != nil {
-		t.Error("execution returned unexpected error")
+		t.Errorf("execution returned unexpected error: %v", err)
 	}
 }
 
 func TestCall(t *testing.T) {
+	// pseudo-Yul: call(0, 0, 0, 0, 0, 0, 34) return(0, msize())
```

Comment:
Why is this pseudo? Probably not balancing the stack? `pop(call(0, 0, 0, 0, 0, 0, 34)) return(0, msize())` would be valid then.

Reply:
Because `pop()` is needed (but not implemented)` and it optimizes it to `DUP1, DUP2, DUP3, ...` instead of `DUP1, DUP1, DUP1, ...` - only `DUP1` is implemented.